### ISSUE TITLE
[Bug]: key "freebsd_amd64" not found in dictionary

### ISSUE THREAD

**Issue by EByrdS** [aspect-build/bazel-lib#541]
### What happened?

I added and `http_archive` that loads this repository transitively. In one of the function calls, it uses [copy_directory_platform_repo](https://github.com/aspect-build/bazel-lib/blob/main/lib/private/copy_directory_toolchain.bzl), but raises the error in the title.

This happens because [COPY_DIRECTORY_PLATORMS](https://github.com/aspect-build/bazel-lib/blob/8fa21275181371b38f76ef30d0aa14006a2b472b/lib/private/copy_directory_toolchain.bzl#L12) has 6 entries, but `COPY_DIRECTORY_INTEGRITY` used [here](https://github.com/aspect-build/bazel-lib/blob/8fa21275181371b38f76ef30d0aa14006a2b472b/lib/private/copy_directory_toolchain.bzl#L170) and declared [here](https://github.com/aspect-build/bazel-lib/blob/main/tools/versions.bzl) has only 5. The missing dictionary entry is `freebsd_amd64`.

### Version

Dev architecture: MacOS arm64

```
bazel --version
bazel 6.3.2-homebrew
```

Tool that I want to install:
```
http_archive(
    name = "aspect_rules_js",
    sha256 = "77c4ea46c27f96e4aadcc580cd608369208422cf774988594ae8a01df6642c82",
    strip_prefix = "rules_js-1.32.2",
    url = "https://github.com/aspect-build/rules_js/releases/download/v1.32.2/rules_js-v1.32.2.tar.gz",
)
```

`aspect_bazel_lib` loaded transitively:
```
http_archive(
    name = "aspect_bazel_lib",
    sha256 = "44f4f6d1ea1fc5a79ed6ca83f875038fee0a0c47db4f9c9beed097e56f8fad03",
    strip_prefix = "bazel-lib-1.34.0",
    url = "https://github.com/aspect-build/bazel-lib/releases/download/v1.34.0/bazel-lib-v1.34.0.tar.gz",
)
```

The error happens when `bazel sync`ing.

The version downloaded there is different from the latest one here, but I think the code would yield the same error given that the key does not exist.

### How to reproduce

```shell
WORKSPACE

http_archive(
    name = "aspect_rules_js",
    sha256 = "77c4ea46c27f96e4aadcc580cd608369208422cf774988594ae8a01df6642c82",
    strip_prefix = "rules_js-1.32.2",
    url = "https://github.com/aspect-build/rules_js/releases/download/v1.32.2/rules_js-v1.32.2.tar.gz",
)


load("@aspect_rules_js//js:repositories.bzl", "rules_js_dependencies")
```

OS: MacOS arm64 (Apple M2 Max)
```


### Any other information?

_No response_

**Comment by alexeagle** [comment]
I don't think we have enough freebsd users to prioritize this ourselves, but we should definitely publish binaries for that platform.
In the meantime you should be able to configure the toolchain registration to build those tools from sources.
